syncd/syncd_applyview.cpp

```diff
@@ -2629,54 +2677,342 @@ std::shared_ptr<SaiObj> findCurrentBestMatchForAclTableGroup(
+            {
+                continue;
+            }
+
+            /*
+             * At this point we have both tunnels which ip mathces
+             */
+
+            if (tmpTunnel->hasAttr(SAI_TUNNEL_ATTR_UNDERLAY_INTERFACE) &&
+                curTunnel->hasAttr(SAI_TUNNEL_ATTR_UNDERLAY_INTERFACE))
+            {
+                auto tmpRif = tmpTunnel->getSaiAttr(SAI_TUNNEL_ATTR_UNDERLAY_INTERFACE);
+                auto curRif = curTunnel->getSaiAttr(SAI_TUNNEL_ATTR_UNDERLAY_INTERFACE);
+
+                if (tmpRif->getSaiAttr()->value.oid == temporaryObj->getVid())
+                {
+                    for (auto c: candidateObjects)
+                    {
+                        if (c.obj->getVid() != curRif->getSaiAttr()->value.oid)
+                            continue;
+
+                        SWSS_LOG_INFO("found best ROUTER_INTERFACE based on TUNNEL underlay interface %s", c.obj->str_object_id.c_str());
+
+                        return c.obj;
+                    }
+                }
+            }
+
+            if (tmpTunnel->hasAttr(SAI_TUNNEL_ATTR_OVERLAY_INTERFACE) &&
+                curTunnel->hasAttr(SAI_TUNNEL_ATTR_OVERLAY_INTERFACE))
+            {
+                auto tmpRif = tmpTunnel->getSaiAttr(SAI_TUNNEL_ATTR_OVERLAY_INTERFACE);
+                auto curRif = curTunnel->getSaiAttr(SAI_TUNNEL_ATTR_OVERLAY_INTERFACE);
+
+                if (tmpRif->getSaiAttr()->value.oid == temporaryObj->getVid())
+                {
+                    for (auto c: candidateObjects)
+                    {
+                        if (c.obj->getVid() != curRif->getSaiAttr()->value.oid)
+                            continue;
 
-        if (candidateLag != nullptr)
-            return candidateLag;
+                        SWSS_LOG_INFO("found best ROUTER_INTERFACE based on TUNNEL overlay interface %s", c.obj->str_object_id.c_str());
+
+                        return c.obj;
+                    }
+                }
+            }
+        }
     }
 
-    if (temporaryObj->getObjectType() == SAI_OBJECT_TYPE_NEXT_HOP_GROUP)
+    SWSS_LOG_NOTICE("failed to find best candidate for LOOPBACK ROUTER_INTERFACE using TUNNEL");
+
+    return nullptr;
+}
+
+std::shared_ptr<SaiObj> findCurrentBestMatchForPolicer(
+        _In_ const AsicView &currentView,
+        _In_ const AsicView &temporaryView,
+        _In_ const std::shared_ptr<const SaiObj> &temporaryObj,
+        _In_ const std::vector<sai_object_compare_info_t> &candidateObjects)
+{
+    SWSS_LOG_ENTER();
+
+    /*
+     * For policer we can see on which hostif trap group is set, and on which
+     * hostif trap.  Hostif trap have SAI_HOSTIF_TRAP_ATTR_TRAP_TYPE attribute
+     * which is KEY and there can be only one trap of that type. we can use
+     * that to match hostif trap group and later on policer.
+     *
+     * NOTE: policer can be set on default hostif trap group. In this case we
+     * are getting processed and not processed objects into account.
```

Comment:
Policer could be applied on port and ACL too.  It looks the logic here limits the processing to hostif only.

Reply:
correct, for current configuration this is only supported, but it will be expanded in the future